test/python/tf_unittest_runner.py

```diff
@@ -39,89 +42,147 @@ def main():
     optional.add_argument(
         '--run_test',
         help=
-        "Runs the testcase and returns the output. Eg:math_ops_test.math_ops_test.DivNoNanTest.testBasic"
+        "Runs the testcase and returns the output. Eg:math_ops_test.DivNoNanTest.testBasic"
     )
     parser._action_groups.append(optional)
     arguments = parser.parse_args()
 
-    all_dirs_to_path(arguments.tensorflow_path)
+    accepted_formats = [
+        "math_ops_test", "mat_ops_test.DivNoNanTest",
+        "math_ops_test.DivNoNanTest.testBasic", "math_ops_test.DivNoNanTest.*",
+        "math_ops_test.D*", "math_ops_test.*", "math_*_test", "math_*_*_test",
+        "math*_test"
+    ]
 
     if (arguments.list_tests):
-        list_tests(arguments.list_tests)
+        try:
+            module_list = regex_walk(arguments.tensorflow_path,
+                                     arguments.list_tests)
+            list_tests(module_list, arguments.list_tests)
+        except:
+            print(
+                "Enter a valid argument to LIST tests.\nList of accepted formats:"
+            )
+            print('\n'.join(accepted_formats))
 
     if (arguments.run_test):
-        run_test(arguments.run_test)
+        try:
+            module_list = regex_walk(arguments.tensorflow_path,
+                                     arguments.run_test)
+            test_list = list_tests(module_list, arguments.run_test)
+            run_test(test_list)
+        except:
+            print(
+                "Enter a valid argument to RUN tests.\nList of accepted formats:"
+            )
+            print('\n'.join(accepted_formats))
 
```

Comment:
`RUN tests.` --> `--run_tests`

Reply:
There is not much difference between the 2 modes, except `run_test(test_list)`

how about this refactor:

```
def helper(tf_path, test_regex):
    try:
            module_list = regex_walk(arguments.tensorflow_path,
                                     arguments.run_test)
            test_list = list_tests(module_list, arguments.run_test)
   except:
        pass #stuff here

if list:
    test_list = helper(arguments.tensorflow_path, arguments.run_test)
    print_tests(test_list)
if run:
    test_list = helper(arguments.tensorflow_path, arguments. list_tests)
   run_test(test_list)
```

of course, in this refactor, `list_tests` would not do the printing of functions